mscore/musescore.cpp

```diff
@@ -5750,6 +5753,49 @@ void MuseScore::transpose()
             cs->deselectAll();
       }
 
+//---------------------------------------------------------
+//   cmdRealizeChordSymbols
+///   Realize selected chord symbols into notes on the staff.
+///   Currently just pops up a dialog to list TPCs,
+///   Intervals, and pitches.
+//---------------------------------------------------------
+
+void MuseScore::realizeChordSymbols()
+      {
+      if (!cs)
+            return;
+      if (!cs->selection().isList()) {
+            QErrorMessage err;
+            err.showMessage("Invalid Selection. Cannot Realize Chord");
+            err.exec();
+            return;
+            }
+      QList<Harmony*> hlist;
+      for (Element* e : cs->selection().elements()) {
+            if (e->isHarmony())
+                  hlist << toHarmony(e);
+            }
+
+      RealizeHarmonyDialog dialog;
+      if (!hlist.empty()) {
+            dialog.setChordList(hlist);
+            }
+      else {
+            QErrorMessage err;
+            err.showMessage("No Chord Selected. Cannot Realize Chord");
+            err.exec();
+            return;
+            }
+
+      if (dialog.exec()) {    //realize the chord symbols onto the track
+            cs->startCmd();
+            cs->cmdRealizeChordSymbols(hlist, dialog.getLiteral(), dialog.optionsOverride() ?
```

Comment:
As an example of the sort of restructuring to consider: do we really need a cmd that takes a list of harmony elements?  Why not have the command assemble the list from the selection?  That seems more consistent with how other commands work, passing lists around seems kludgy to me.  Maybe there is a good reason here, anyhow, this is the sort of thing we'll want to take a look at.

Reply:
Ah yes that is a change to make. Passing the list was a bit of an artifact from when I was trying to allow the dialog to manipulate the list of symbols to realize. I scrapped that idea though after I realized that it wouldn't be the most useful feature and would probably just add more UI clutter since the user can just modify the selection itself.